pkg/api/login_oauth.go

```diff
@@ -218,6 +218,10 @@ func (hs *HTTPServer) OAuthLogin(ctx *m.ReqContext) {
 
 	if redirectTo, _ := url.QueryUnescape(ctx.GetCookie("redirect_to")); len(redirectTo) > 0 {
 		middleware.DeleteCookie(ctx.Resp, "redirect_to", hs.cookieOptionsFromCfg)
+		// if there is a sub_url, prepend it
+		if setting.AppSubUrl != "" {
+			redirectTo = setting.AppSubUrl + redirectTo
+		}
```

Comment:
This change doesn't work for me since redirect cookie already contains subpath so resulting redirect is `http://localhost:3000/grafana/grafana/d/5SdHCadmz/panel-tests-graph?orgId=1`. Removing this code makes it work for as expected for me.

Reply:
will investigate, i was testing this on a "real" site behind SSL and nginx, but not testing locally.

nginx block looks like this, the rewrite is probably "fixing" that for me
```
  upstream internal {
    server 0.0.0.0:9098;
    keepalive 32;
  }
  location /grafana/ {
      proxy_pass http://internal;
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";
      proxy_set_header Host $http_host;
      rewrite ^/grafana/(.*) /$1 break;
  }
```

Reply:
have removed this change, and verified it is working.  the redirect_to subpath is handled in login.go